### ISSUE TITLE
crypto.go bug

### ISSUE THREAD

**Issue by tgulacsi** [pdfcpu/pdfcpu#182]
https://github.com/pdfcpu/pdfcpu/blob/6d5c8c0503790560df905e9bed9aff8b59232e79/pkg/pdfcpu/crypto.go#L759
`if eff != nil && *strf != "Identity"` instead of `if eff != nil && *eff != "Identity"`

https://github.com/pdfcpu/pdfcpu/blob/6d5c8c0503790560df905e9bed9aff8b59232e79/pkg/pdfcpu/crypto.go#L766
`*strf` instead off `*eff`

**Comment by hhrutter** [comment]
Can you please do me the favour and describe the issue are running into?
Issues are meant as an informative reference for all pdfcpu users.
Thank you!

**Comment by hhrutter** [comment]
OK. This is indeed a nasty bug!
Thanks for reporting this.

**Comment by tgulacsi** [comment]
Just took a stab at getting rid of pointers, and during the conversion it came up as suspicious code.

**Comment by hhrutter** [comment]
Are you doing that across the board in your fork?
What are you gaining by doing that?

**Comment by tgulacsi** [comment]
No, I just try to avoid pointers if possible - I hate panics, esp. in library code, as it brings down my app, instead of just returning an error!
When I need to distinguish between a 0 and a nil, I use what sql.NullInt64 does: augment it with a bool.

Now I see that getting rid of all those *string, *int and *int64 is not trivial, a few full days of work, with nothing early reward, esp. if won't be accepted upstream.

Until pdfcpu won't panic, I have to wrap it in recover().

**Comment by hhrutter** [comment]
I also hate panics, so you do have a point.
Still, there were numerous careful considerations to go with string pointers.
It is some time ago, but there were a couple of good reasons where I needed string pointers.
For consistency I applied this decision across the board.
Another thing is even by avoiding string pointers you don't get rid of all possible panic situations.
There are of course other pointers in structs and even the stdlib panics sometime.

I was not aware that you are planning a PR for this.
This is a major issue and I need to think where I am going with this long term, so it is not a good PR candidate but the issue and the solution (reference to sql.NullInt64) is very valid and duly noted.

**Comment by tgulacsi** [comment]
Now I turned down the idea of a PR for this - it's too much undertaking.

I've started the conversion, just to see how intrusive would that be.
It seems that it's absolutely everywhere, and I don't have the knowledge, that where do we have to differentiate a 0 from a nil, especially in  the structs.

So maybe some smaller steps (only get rid of *strings, then later maybe *int and *int64) would work better.
